src/Controller.cpp

```diff
@@ -357,6 +270,11 @@ void Controller::reset(const mc_control::ControllerResetData & data)
 
 void Controller::internalReset()
 {
+  // FIXME:
+  // - resets does not respect the foot plan position when pausing walking (feet come back align
+  // dragging on the floor)
+  // - Do we still need the posture tricks?
```

Comment:
Looks like the answer is yes, that's still how the controller implements [Chest posture control (Section III.B)](https://staff.aist.go.jp/k.kaneko/publications/2010_publications/IROS2010-0872.pdf). Without it, ankle compliance will comply uncompensated to the hip as we discussed last year https://github.com/stephane-caron/lipm_walking_controller/issues/32, and that's extra work for the stabilizer (will likely fail the plank test then).

The indirect implementation via the anchor frame is not good, it's unnecessarily obscure. It is much better to do straight explicit feedback on the pelvis orientation like Kajita _et al._ proposed.

Reply:
Thanks for the detailed description of the role of pelvis control.

Here I was more referring to the fact that you are resetting the reference posture at every single iteration. For LIPMWalking itself this is not an issue, however many people have made custom controllers based on this one, and when used in other contexts such as manipulation, or when one explicitely wants to looks somewhere this becomes annoying. It also interferes with some FSM behaviours, such as the Interrupt button in the FSM (the robot comes back to halfsitting instead of keeping its current posture). Anyways I'll change the logic of this in a subsequent PR.